fix(curriculum): update React import/export lesson to cover default and named exports.

StephenMuya opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitHub Codespaces.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #60106

<!-- Feel free to add any additional description of changes below this line -->

StephenMuya committed:
chore(lesson): update React import/export lesson to cover default and named exports

majestic-owl448 reviewed (commented):
"default export" and "named export" are not code snippets, please don't write them in backticks, same for "import" and "export" when used as a verb

StephenMuya committed:
update lecture title for Introduction to JavaScript libraries and frameworks

majestic-owl448 commented on curriculum/challenges/english/blocks/lecture-introduction-to-javascript-libraries-and-frameworks/674ba6876f7ada867135bb95.md:
```diff
@@ -9,6 +9,10 @@ dashedName: how-can-you-import-and-export-components-in-react
 
+An export makes a component available to other files. An import allows a file to use a component that was exported elsewhere. In React projects, this is how components are reused and combined to build user interfaces.
+
+In this lesson, you will learn how to export React components using `default exports` and `named exports`, and how to import them where they are needed.
```
here there are still backticks to remove

majestic-owl448 reviewed (commented):
this is optional, but do you maybe want to update or replace one of the questions to also be about named exports?

majestic-owl448 reviewed (commented):
please do not force push unless absolutely necessary, some review comments got lost. let me add them again...

StephenMuya committed:
add named export assesment questions to react lesson.

gikf commented on curriculum/challenges/english/blocks/lecture-introduction-to-javascript-libraries-and-frameworks/674ba6876f7ada867135bb95.md:
```diff
@@ -258,3 +337,102 @@ Remember that you learned how to work with imports in earlier lessons.
+
+How would you export multiple components from the same file using named exports?
+
+## --answer--
```
```suggestion
## --answers--
```
Causes linting error.

StephenMuya committed:
correct section title from "answer" to "answers".

majestic-owl448 commented on curriculum/challenges/english/blocks/lecture-introduction-to-javascript-libraries-and-frameworks/674ba6876f7ada867135bb95.md:
```diff
@@ -258,3 +337,102 @@ Remember that you learned how to work with imports in earlier lessons.
+
+Remember that a JavaScript module is limited to only one default export.
+
+## --video-solution--
```
missing solution number

majestic-owl448 reviewed (commented):
we usually out three questions for each lesson, for that reason I asked you to replace one of the questions not add more. Can you make so that we have only three questions?

also please make sure to write under `video-solution` the number of the solution, and not leave it empty

StephenMuya committed:
update export questions to focus on named exports in React.

StephenMuya committed:
Add configuration and utility functions for curriculum and donation settings

- Implement tests for curriculum super blocks and immutability checks.
- Create donation settings configuration for payment providers and subscription plans.
- Add internationalization support with language codes and names.
- Introduce utility functions for line manipulation and validation checks.
- Develop polyvinyl utilities for file handling and transformations.
- Implement tests for validation functions and polyvinyl utilities.
- Add shuffle array utility for randomizing array elements.

StephenMuya committed:
Add video solution number for named export question.

majestic-owl448 committed:
Merge branch 'main' into react-import-export-update

majestic-owl448 reviewed (approved):
LGTM 🎈 

branch updated to include recent breaking changes
